WordPress/gutenberg
# Try: make logic for conditional editor_styles consistent

## What?

In https://github.com/WordPress/gutenberg/pull/44640, we added a check for `wp-block-styles` theme support before including the `wp-block-library-theme` CSS in the editor.

This seemed to work well in most cases, however, there are still some cases where this CSS is included. As seen in this issue for TT3 https://github.com/WordPress/twentytwentythree/issues/295, the opinionated width of 100px for the separator is still being loaded from the theme.css file in the editor, but this isn't used on the front end.

| Editor | Front end |
| ----- | ---------- |
| <img width="398" alt="image" src="https://user-images.githubusercontent.com/1645628/198323197-f643c79b-e18d-4b06-92e3-74157cd77f19.png"> | <img width="485" alt="image" src="https://user-images.githubusercontent.com/1645628/198323383-ea7aaaca-767e-443a-9be9-bd0ee9cb8102.png"> |

## Why?

The editor styles should match the front end.

## How?

This PR is an attempt to never allow the `wp-block-library-theme` CSS to be loaded under the conditions we introduced in https://github.com/WordPress/gutenberg/pull/44640.

I've made the logic consistent between /template-parts-screen.php and /script-loader.php, so the condition to include the theme.css file is only if the active theme supports `wp-block-styles` AND there are no `editor_styles` set.

## Testing Instructions




1. Activate TT3 / a theme that doesn't support `wp-block-styles` AND doesn't include `add_editor_style( 'style.css' );`
2. Insert a Separator block, see the difference in width between the editor and front end

Example separator markup for copy/pasting:

```

<hr class="wp-block-separator has-alpha-channel-opacity"/>

```
